starter-code/scripts/article.js

```diff
@@ -50,13 +50,20 @@ Article.fetchAll = function() {
     // When rawData is already in localStorage,
     // we can load it with the .loadAll function above,
     // and then render the index page (using the proper method on the articleView object).
-    Article.loadAll(); //TODO: What do we pass in to loadAll()?
-    //TODO: What method do we call to render the index page?
+    Article.loadAll(JSON.parse(localStorage.rawData)); //DONE: What do we pass in to loadAll()?
+    //DONE: What method do we call to render the index page?
+    articleView.initIndexPage();
   } else {
-    // TODO: When we don't already have the rawData,
+    // DONE!: When we don't already have the rawData,
     // we need to retrieve the JSON file from the server with AJAX (which jQuery method is best for this?),
     // cache it in localStorage so we can skip the server call next time,
     // then load all the data into Article.all with the .loadAll function above,
     // and then render the index page.
+    // Est time: 15 mins | Done in: 20 mins
+    $.getJSON('data/hackerIpsum.json').then(
+      function(data) {
+        localStorage.rawData = JSON.stringify(data);
+        Article.fetchAll();},
```

Comment:
Why are you calling Article.fetchAll inside Article.fetchAll? If we look at the if statement above, what are we calling there? Should this also be called here? Hint: yes :)